File: 0058-length-of-last-word/0058-length-of-last-word.js
public class Solution {
    public int LengthOfLastWord(string s) {
        int length = 0;
        int pos = s.Length - 1;

        // Skip trailing spaces if any
        while (pos >= 0 && s[pos] == ' ') {
            --pos;
        }

        // Count characters in the last word
        while (pos >= 0 && s[pos] != ' ') {
            --pos;
            ++length;
        }

        return length;           
    }
}

File: 0058-length-of-last-word/0058-length-of-last-word.rs
impl Solution {
	pub fn length_of_last_word(s: String) -> i32 {
		let mut word = String::new();
		for ch in s.chars().rev() {
			if ch == ' '{
				if word.len() != 0 {
					break;
				}
			} else {
				word.push(ch)
			}
		}
		word.len() as _
	}
}